components/Quiz.tsx

```diff
@@ -0,0 +1,95 @@
+"use client"
+
+import { Button } from "@shadcn/ui/button"
+import { Checkbox } from "@shadcn/ui/checkbox"
+import { Label } from "@shadcn/ui/label"
+import { RadioGroup, RadioGroupItem } from "@shadcn/ui/radio-group"
+
+import { Typography } from "./Typography"
+
+type Answer = {
+  text: string
+  isCorrect: boolean
+}
+
+interface Props {
+  question: string
+  answers: Answer[]
+}
+
+export function Quiz({ question, answers }: Props) {
+  const isMultipleCorrect = answers.filter((answer) => answer.isCorrect).length > 1
+
+  async function onSubmit(formdata: FormData) {
+    const selectedAnswers = Array.from(formdata.getAll("answer"))
+
+    /**
+     * Return if no value is selected
+     */
+    if (selectedAnswers.length < 1) return
+
+    const correctAnswers = answers.filter((answer) => answer.isCorrect).map((answer) => answer.text)
+
+    /**
+     * If answers are more than 1.
+     */
+    if (isMultipleCorrect) {
+      const corretAnswerCount = correctAnswers.length
+      const selectedAnswerCount = selectedAnswers.length
+
+      if (
+        corretAnswerCount === selectedAnswerCount &&
+        correctAnswers.every((answer) => selectedAnswers.includes(answer))
+      ) {
+        alert("Correct!")
+      } else {
+        alert("Incorrect!")
+      }
+      return
```

Comment:
**nitpick:** this return shouldn't be necessary, because we have if/else.

Reply:
Awesome :)

Reply:
**bug**: We don't get the `incorrect` alert when multiple questions are submitted (for checkboxes)

Reply:
I will take a look at it once I am back to Stable internet.